### ISSUE TITLE
Update version (+ on pip / Anaconda)

### ISSUE THREAD

**Issue by asross** [pyqg/pyqg#277]
I've gotten some requests to have the newest version of pyqg that supports parameterizations installable via pip / conda. Any chance we could tag and upload a new version?

**Comment by rabernat** [comment]
Hi @asross! Yes this is a good idea. Unfortunately it is not as easy as it should be - see #255.

If you would like to take the lead on the release, please be my guest. I have added you to the maintainers group @pyqg/pyqg_devs.

**Comment by asross** [comment]
Awesome. I should warn that I haven't done too much package publishing in Python (mostly Ruby/JS), so migrating from versioner to setuptools_scm might be a bit tricky -- especially in a way that doesn't break cython or FFTs. So if there is a some kind of existing process for publishing a new package without that overhaul, that would be great. But I'll look into it if not!

**Comment by rabernat** [comment]
For now it is possible to do the release manually. Instructions are here: https://pyqg.readthedocs.io/en/latest/development.html#release-procedure

**Comment by asross** [comment]
Sorry for the slow progress on this -- the first thing I noticed was that the release instructions haven't been updated since versioneer was added, so the way to increment the version number now seems to be to just add the new git tag (`v0.5.0`) to the appropriate commit (and versioneer appears to be able to automatically infer the version number from the tag?).

The second thing I noticed, however, was that `master` currently contains some of my changes to `calc_ispec` (merged after the parameterization PR merge) which, per https://github.com/pyqg/pyqg/issues/275, now seem incorrect. So I was thinking of trying to resolve those first before incrementing the version. However, figuring out the right normalization factor has proved kind of tricky.

My pyqg plan now is to instead:
- assign the 0.5.0 tag to https://github.com/pyqg/pyqg/tree/00990ba51c85c9c2ad3b066ef166154c27ca70bf (when the parameterization PR was merged)
- run through the rest of the release process (i.e. pypi sdist, assuming you adding me to the maintainers group on Github gave me permission to update pypi)
- create a separate PR to update the release instructions and whats-new.rst
- hopefully move on to figuring out calc_ispec normalization and getting that merged (then updating to 0.5.1)
- hopefully move on from there to creating some additional PRs with diagnostics refactors/units and new features :)

**Comment by rabernat** [comment]
Sounds like a great plan @asross! Let me know if you get blocked on anything.

I just added you as an owner on pypi for pyqg.

**Comment by asross** [comment]
Unfortunately, getting the new package to work requires first resolving https://github.com/pyqg/pyqg/issues/216, which so far has resisted [a number of attempts](https://test.pypi.org/project/pyqg/#history).

I'm leaning towards removing Versioneer and reverting to the older, much simpler build process that requires manually setting the version string, if that's ok. The only thing Versioneer provides is automatic association of tags and version numbers, which doesn't seem worth the difficulty.

**Comment by rabernat** [comment]
Yes that is fine with me. Versioneer is unmaintained.

Setuptools_scm is the preferred solution these days.

**Comment by asross** [comment]
Making progress on this, albeit slowly -- https://github.com/pyqg/pyqg/pull/278 is mostly there (modulo fixing the docs and CI), but we've got some decisions to make about whether pyfftw should be the default (which also affect CI).
